solarforecastarbiter/io/reference_observations/common.py

```diff
@@ -303,3 +310,109 @@ def site_name_no_network(site):
+        A site object.
+    template_forecast : solarforecastarbiter.datamodel.Forecast
+        A Forecast object that will only have name, site, variable, and
+        issue_time_of_day replaced. New keys may be added to extra parameters.
+    variable : string
+        Variable measured in the forecast.
+
+    Returns
+    -------
+    created
+        The datamodel object of the newly created forecast.
+    """
+    extra_parameters = json.loads(template_forecast.extra_parameters)
+    if piggyback_on is not None:
+        extra_parameters['piggyback_on'] = piggyback_on
+    site_name = site_name_no_network(site)
+    fx_name = f'{site_name} {template_forecast.name} {variable}'
+    # Some site names are too long and exceed the API's limits,
+    # in those cases. Use the abbreviated version.
+    if len(fx_name) > 64:
+        try:
+            site_extra_parameters = decode_extra_parameters(site)
+        except ValueError:
+            fx_name = fx_name[:64]
+            logger.warning("Forecast name truncated to %s", fx_name)
+        else:
+            site_abbreviation = site_extra_parameters[
+                "network_api_abbreviation"]
+            fx_name = f'{site_abbreviation} {template_forecast.name} {variable}'
+
+    # adjust issue_time_of_day to localtime for standard time, not DST
+    issue_time_of_day = pd.Timestamp.combine(
+        dt.date(2019, 2, 1), template_forecast.issue_time_of_day,
+        ).tz_localize(site.timezone).tz_convert('UTC').time()
+
+    forecast = template_forecast.replace(
+        name=fx_name, extra_parameters=json.dumps(extra_parameters),
+        site=site, variable=variable, issue_time_of_day=issue_time_of_day,
+    )
+    existing = existing_forecasts(api)
+    if forecast.name in existing:
+        logger.info('Forecast, %s, already exists', forecast.name)
+        return existing[forecast.name]
+
+    try:
+        created = api.create_forecast(forecast)
+    except HTTPError as e:
+        logger.error(f'Failed to create {variable} forecast at Site '
+                     f'{site.name}.')
+        logger.debug(f'HTTP Error: {e.response.text}')
+    else:
+        logger.info(f"{created.name} created successfully.")
+        return created
+
+
+def create_forecasts(api, site, variables):
+    """Create Forecast objects for each of variables, if NWP forecasts
+    can be made for that variable. Each of TEMPLATE_FORECASTS will be
+    updated with the appropriate parameters for each variable. Forecasts
+    will also be grouped together via 'piggyback_on'.
+
+    Parameters
+    ----------
+    api : io.APISession
+        An APISession with a valid JWT for accessing the Reference Data user.
+    site : solarforecastarbiter.datamodel.site
+        A site object.
+    variables : list-like
+        List of variables to make a new forecast for each of TEMPLATE_FORECASTS
+    """
+    vars_ = set(variables)
+    diff = vars_ - CURRENT_NWP_VARIABLES
+    if diff:
```

Comment:
how would this happen in practice?

Reply:
for example, were getting relative humidity and poa observations for the RTC sites, but right now the NWP forecast generation doesn't produce those variables